src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```
@@ -142,7 +153,7 @@ public boolean isStorageConnected() {
   @Override
   public boolean addCluster(Cluster cluster) {
     try {
-      session.execute(insertClusterPrepStmt.bind(cluster.getName(), cluster.getPartitioner(), cluster.getSeedHosts()));
+      session.executeAsync(insertClusterPrepStmt.bind(cluster.getName(), cluster.getPartitioner(), cluster.getSeedHosts()));
```

It looks like you're not waiting for any async write to return and exit the method immediatly.
This will lead to potential silent failures and chained calls being inconsistent (we may try to read something that isn't yet written to all replicas).

Is it on purpose or did you forget the `.get()` after the `executeAsync` calls ? 
Also, in the case we have a single call in the method, we may be better just running a synchronous `execute`.

---

It was on purpose.

> This will lead to potential silent failures and chained calls being inconsistent (we may try to read something that isn't yet written to all replicas).

Can you describe any situations where and how this would happen?

I don't know the code thoroughly, but from what i've seen so far:
 - within this request the `cluster` object exists in-memory, and is used as such,
 - whether a separate request doesn't yet find it is part of the already existing eventual consistency of the design.

---

> This will lead to potential silent failures and chained calls being inconsistent.

There is a risk of the async request silently failing. But as an idempotent request we expect it to retry.

The failure mode is interesting though. Already before this PR, there's no guarantee that the request completes (the jvm could crash, and/or it fails and an exception is thrown (and maybe it worked??).